reactor-core/src/main/java/reactor/core/publisher/FluxReplay.java

```diff
@@ -1142,10 +1143,15 @@ public void onError(Throwable t) {
 				for (ReplaySubscription<T> rs : a) {
```

Comment:
Which test does it break? it seems that doing that synchronization after is open to concurrent subscribe(Subscriber) and EMPTY termination.